### PR TITLE
Doxygen updates

### PR DESCRIPTION
**Pull request by yossi2le**
### Description

Doxygen updates for kvstore and DeviceKey

### CODE REVIEW

File: features/storage/kvstore/filesystemstore/FileSystemStore.h [hunk 1]
```diff
@@ -180,6 +185,7 @@ class FileSystemStore : public KVStore {
 
     /**
      * @brief Start an iteration over FileSystemStore keys.
+     *        There are no issue with any other operation while iterator is open.
```
Comment (theamirocohen) [inline features/storage/kvstore/filesystemstore/FileSystemStore.h]: There are no **issues** with any other **operations** while iterator is open.

File: features/storage/kvstore/filesystemstore/FileSystemStore.h [hunk 2]
```diff
@@ -192,6 +198,7 @@ class FileSystemStore : public KVStore {
 
     /**
      * @brief Get next key in iteration.
+     *        There are no issue with any other operation while iterator is open.
```
Comment (theamirocohen) [inline features/storage/kvstore/filesystemstore/FileSystemStore.h]: the same here

File: features/storage/kvstore/global_api/kvstore_global_api.h [hunk 3]
```diff
@@ -95,7 +95,8 @@ int kv_remove(const char *full_name_key);
 
 /**
  * @brief Start an iteration over KVStore keys to find all the entries
- *        that fit the full_prefix
+ *        that fit the full_prefix. There are no issue with any other operation while
```
Comment (theamirocohen) [inline features/storage/kvstore/global_api/kvstore_global_api.h]: There are no **issues** with any other **operations** while iterator is open.

File: features/storage/kvstore/global_api/kvstore_global_api.h [hunk 4]
```diff
@@ -109,7 +110,8 @@ int kv_remove(const char *full_name_key);
 int kv_iterator_open(kv_iterator_t *it, const char *full_prefix);
 
 /**
- * @brief Get next key in iteration that matches the prefix.
+ * @brief Get next key in iteration that matches the prefix. There are no issue with any
+ *        other operation while iterator is open.
```
Comment (theamirocohen) [inline features/storage/kvstore/global_api/kvstore_global_api.h]: the same

File: features/storage/kvstore/kv_map/KVMap.h [hunk 5]
```diff
@@ -220,9 +221,9 @@ class KVMap : private mbed::NonCopyable<KVMap> {
     /**
      * @brief Full name lookup, and then break it into KVStore config and key
      *
-     * @param full_name  String parameter contains the /partition name/key.
-     * @param kv_config The configuration struct associated with the partition name
-     * @param key_index An index to the first character of the key.
+     * @param[in] full_name  String parameter contains the /partition name/key.
+     * @param[out] kv_config Returns, The configuration struct associated with the partition name
+     * @param[out] key_index Returns, an index to the first character of the key.
```
Comment (theamirocohen) [inline features/storage/kvstore/kv_map/KVMap.h]: No commas needed

File: features/storage/kvstore/tdbstore/TDBStore.h [hunk 6]
```diff
@@ -198,6 +205,7 @@ class TDBStore : public KVStore {
 
     /**
      * @brief Start an iteration over KVStore keys.
+     *        There are no issue with any other operation while iterator is open.
```
Comment (theamirocohen) [inline features/storage/kvstore/tdbstore/TDBStore.h]: There are no **issues** with any other **operations** while iterator is open.

File: features/storage/kvstore/tdbstore/TDBStore.h [hunk 7]
```diff
@@ -210,6 +218,7 @@ class TDBStore : public KVStore {
 
     /**
      * @brief Get next key in iteration.
+     *        There are no issue with any other operation while iterator is open.
```
Comment (theamirocohen) [inline features/storage/kvstore/tdbstore/TDBStore.h]: the same

File: features/storage/kvstore/tdbstore/TDBStore.h [hunk 8]
```diff
@@ -237,7 +246,8 @@ class TDBStore : public KVStore {
     virtual int iterator_close(iterator_t it);
 
     /**
-     * @brief Set data in reserved area.
+     * @brief Set data in reserved area which is a special location for special data such as ROT.
+     *        The data written to reserved area can not be overwrite.
```
Comment (theamirocohen) [inline features/storage/kvstore/tdbstore/TDBStore.h]: can not be overwrite **overwritten.**

File: features/storage/kvstore/filesystemstore/FileSystemStore.h [hunk 9]
```diff
@@ -85,7 +88,7 @@ class FileSystemStore : public KVStore {
     virtual int set(const char *key, const void *buffer, size_t size, uint32_t create_flags);
 
     /**
-      * @brief Get one FileSystemStore item, given key.
+      * @brief Get one FileSystemStore item by given key.
```
Comment (dannybenor) [inline features/storage/kvstore/filesystemstore/FileSystemStore.h:91]: MBED_ERROR_INVALID_DATA_DETECTED    Data is corrupt. ==>corrupted

**Review by dannybenor — APPROVED** [review]

File: features/storage/kvstore/filesystemstore/FileSystemStore.h (hunk 9 above)
Comment (offirko) [reply features/storage/kvstore/filesystemstore/FileSystemStore.h]: @yossi2le - Thanks for updates and fixes , LGTM

**Comment by offirko** [comment]
@yossi2le - Thanks for updates and fixes , LGTM

**Pull request closed without merging**